orc8r/cloud/go/services/orchestrator/servicers/builder_servicer.go

```diff
@@ -34,37 +36,42 @@ import (
 	mconfig_protos "magma/orc8r/lib/go/protos/mconfig"
 )
 
-var localBuilders = []mconfig.Builder{
-	&baseOrchestratorBuilder{},
+type builderServicer struct {
+	orc8Version string
 }
 
-type builderServicer struct{}
-
-func NewBuilderServicer() builder_protos.MconfigBuilderServer {
-	return &builderServicer{}
+func NewBuilderServicer(orc8rVersion *string) builder_protos.MconfigBuilderServer {
```

Comment:
I would personally prefer separating this out into two separate functions, as discussed, that way 1 is "always stateless" and the other is "always stateful". In this hybrid impl, its statefulness requires understanding/thinking through the implementation

Reply:
I personally believe that having single point of entry makes it easy to follow the code flow(which is very useful when debugging a time-sensitive issue), but i don't have very strong feelings about this one, so i can change it.

Reply:
There's still 1 flow, so the conceptual complexity will be ~same. Separate is preferable if you're amenable. Here's an example of where we currently do this in the codebase: https://github.com/magma/magma/blob/13b485a9b6da5874761919c840e4c0a8be81641d/orc8r/cloud/go/obsidian/swagger/spec_servicer.go#L49-L52

Reply:
since go doesnt allow fn overloading i will have to use different names, so now some trying to debug will have two starting points, depending on whether they provide orc8r version or not. but like i said, i dont have strong feelings about it, so i changed it